dnf/rpm/__init__.py

```diff
@@ -45,6 +45,17 @@ def detect_releasever(installroot):
             msg = 'Error: rpmdb failed to list provides. Try: rpm --rebuilddb'
             raise dnf.exceptions.Error(msg)
         releasever = hdr['version']
+        try:
+            off = hdr[getattr(rpm, 'RPMTAG_PROVIDENAME')].index(distroverpkg)
+            flag = hdr[getattr(rpm, 'RPMTAG_PROVIDEFLAGS')][off]
```

Comment:
`hdr[rpm.RPMTAG_PROVIDEFLAGS]` or `hdr['provideflags']`, no need for `getattr()`. Same for above and below.